### PR TITLE
[SPARK-7690][ML] Multiclass classification Evaluator

### PR DESCRIPTION
**Pull request by harsha2010**
Multiclass Classification Evaluator for ML Pipelines. F1 score, precision, recall, weighted precision and weighted recall are supported as available metrics.

### CODE REVIEW

**Comment by jkbradley** [comment]
Are you aiming this for 1.5?  Let me know when it's ready for review.

**Comment by harsha2010** [comment]
@jkbradley this PR is ready for an initial review. I have added f1 score (the weighted f measure) and named it such as it is familiar from scikit learn. I'll add a couple of the other dominant metrics as well, I am thinking accuracy and kappa similar to what caret package in R provides.
suggestions on what other metrics are higher priority?

**Comment by jkbradley** [comment]
OK, I'll take a look.  Calling f1 "f1" sounds good to me.  Since we're pressed for time for 1.5, I'd say we should add whatever is easily available from MulticlassMetrics but nothing else for now.

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.apache.spark.ml.param.{ParamMap, ParamValidators, Param}
+import org.apache.spark.ml.param.shared.{HasLabelCol, HasPredictionCol}
+import org.apache.spark.ml.util.{SchemaUtils, Identifiable}
+import org.apache.spark.mllib.evaluation.MulticlassMetrics
+import org.apache.spark.sql.{Row, DataFrame}
+import org.apache.spark.sql.types.DoubleType
+
+/**
+ * :: Experimental ::
+ * Evaluator for binary classification, which expects two input columns: score and label.
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: update doc

File: python/pyspark/ml/evaluation.py [hunk 2]
```diff
@@ -214,6 +214,65 @@ def setParams(self, predictionCol="prediction", labelCol="label",
...
 
+
+@inherit_doc
+class MulticlassClassificationEvaluator(JavaEvaluator, HasLabelCol, HasPredictionCol):
+    """
+    Evaluator for Multiclass Classification, which expects two input
+    columns: prediction and label.
+    >>> scoreAndLabels = [(0.0, 0.0), (0.0, 1.0), (0.0, 0.0),
+    ...     (1.0, 0.0), (1.0, 1.0), (1.0, 1.0), (1.0, 1.0), (2.0, 2.0), (2.0, 0.0)]
+    >>> dataset = sqlContext.createDataFrame(scoreAndLabels, ["raw", "label"])
```
Comment (jkbradley) [inline python/pyspark/ml/evaluation.py]: nit: Rename "raw" to "prediction"

**Comment by jkbradley** [comment]
It looks good; I don't see any real problems.  Let me know when to make a final pass.

Note: You could add a very basic unit test for Scala, copying the BinaryClassificationEvaluator one.

**Comment by harsha2010** [comment]
@jkbradley added tests and additional metrics like precision, recall, weightedPrecision and weightedRecall. Agreed we can wait on additional metrics not already contained in MulticlassMetrics.
Ready for final pass.

**Comment by jkbradley** [comment]
LGTM.  I'll merge this with master.  Thank you!

**Comment by jkbradley** [comment]
Oops, I just realized: Can you please change the "mllib" tag to "ml" in the title and add a description in the first PR comment?  Trying to be better about this...

**Comment by harsha2010** [comment]
@jkbradley done, thanks!

**Pull request closed without merging**